From d186856ba1914ca0fbf715fe9b0f31067dd517a4 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Thu, 19 Sep 2019 11:30:13 +0800
Subject: [PATCH 310/767] mmc: cavium: Drop the aligned check for the dma
 address

In commit 73c1489f17e8 ("mmc: cavium: forbid unaligned DMA"), the codes
are added to check the unaligned dma request. But at that time,
we still don't do the dma map for the scatterlist yet. So it is
meaningless to check the dma address. So drop of it.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 drivers/mmc/host/cavium.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index fc6464957df9..d724155d7eb6 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -853,7 +853,7 @@ static void cvm_mmc_dma_request(struct mmc_host *mmc,
 	/* unaligned multi-block DMA has problems, so forbid all unaligned */
 	for (seg = 0; seg < mrq->data->sg_len; seg++) {
 		struct scatterlist *sg = &mrq->data->sg[seg];
-		u64 align = (sg->offset | sg->length | sg->dma_address);
+		u64 align = (sg->offset | sg->length);
 
 		if (!(align & 7))
 			continue;
-- 
2.31.1

